### PR TITLE
Support watching several shadow-cljs builds.

### PR DESCRIPTION
**Pull request by ageneau**
Support watching several shadow-cljs builds.

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`eldev test`)
- [x] All code passes the linter (`eldev lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [x] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

Thanks!

*If you're just starting out to hack on CIDER you might find this [section of its
manual][1] extremely useful.*

[1]: https://docs.cider.mx/cider/contributing/hacking.html

### CODE REVIEW

**Comment by ageneau** [comment]
The linter check fails but it's not related to anything I changed I believe.

File: cider.el [hunk 1]
```diff
@@ -719,6 +719,11 @@ Generally you should not disable this unless you run into some faulty check."
       options
     (concat ":" options)))
 
+(defcustom cider-shadow-watched-builds nil
+  "Defines the list of builds `shadow-cljs' should watch."
+  :type '(repeat string)
+  :package-version '(cider . "0.26.2"))
```
Comment (bbatsov) [inline cider.el:725]: This should say "1.0".

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### New features
 
 * [#2909](https://github.com/clojure-emacs/cider/issues/2909): Add new customization variable `cider-inspector-auto-select-buffer` to control the auto selection of the inspector buffer.
+* Add a new customization variable cider-shadow-watched-builds to allow watching several shadow-cljs builds at the same time.
```
Comment (bbatsov) [inline CHANGELOG.md:8]: You can link to your PR here.

**Comment by ageneau** [comment]
PR updated with your feedback.

**Pull request merged**